js/util/ajax.js

```diff
@@ -59,7 +61,11 @@ exports.getImage = function(url, callback) {
             (window.URL || window.webkitURL).revokeObjectURL(img.src);
         };
         var blob = new window.Blob([new Uint8Array(imgData)], { type: 'image/png' });
-        img.src = (window.URL || window.webkitURL).createObjectURL(blob);
+        if (imgData.byteLength) {
```

Comment:
According to the w3c spec, a 204 response may not include any body

> ## 10.2.5 204 No Content
> ...
> 
> The 204 response MUST NOT include a message-body, and thus is always terminated by the first empty line after the header fields.

My intuition is that we should keep the behaviour you have implemented in response to a 204 header and fail loudly if any other type of response contains no data.